tests/gui/os/window-progress.spec.js

```diff
@@ -27,41 +27,82 @@ describe('Browser: WindowProgress', function() {
 
       beforeEach(function() {
         this.setProgressBarSpy = m.sinon.spy();
+        this.setTitleSpy = m.sinon.spy();
 
         windowProgress.currentWindow = {
-          setProgressBar: this.setProgressBarSpy
+          setProgressBar: this.setProgressBarSpy,
+          setTitle: this.setTitleSpy
+        };
+
+        this.state = {
+          percentage: 85,
+          speed: 100,
+          type: 'write'
         };
       });
 
       describe('.set()', function() {
 
         it('should translate 0-100 percentages to 0-1 ranges', function() {
-          windowProgress.set(85);
+          windowProgress.set(this.state);
           m.chai.expect(this.setProgressBarSpy).to.have.been.calledWith(0.85);
         });
 
         it('should set 0 given 0', function() {
-          windowProgress.set(0);
+          this.state.percentage = 0;
+          windowProgress.set(this.state);
           m.chai.expect(this.setProgressBarSpy).to.have.been.calledWith(0);
         });
 
         it('should set 1 given 100', function() {
-          windowProgress.set(100);
+          this.state.percentage = 100;
+          windowProgress.set(this.state);
           m.chai.expect(this.setProgressBarSpy).to.have.been.calledWith(1);
         });
 
         it('should throw if given a percentage higher than 100', function() {
+          this.state.percentage = 101;
+          const state = this.state;
           m.chai.expect(function() {
-            windowProgress.set(101);
+            windowProgress.set(state);
           }).to.throw('Invalid percentage: 101');
         });
 
         it('should throw if given a percentage less than 0', function() {
+          this.state.percentage = -1;
+          const state = this.state;
```

Comment:
I've just noticed that some unit-tests do:
```
this.state.foo = blah;
windowProgress.set(this.state);
```
but others do:
```
this.state.foo = blah;
const state = this.state;
windowProgress.set(state);
```

Any reason for that?